resources/js/tryitout.js

```diff
@@ -51,6 +51,17 @@ function cancelTryOut(endpointId) {
     }
 }
 
+function recursiveAddSearchParams(url, previousAppend, previousKey, obj) {
```

Comment:
Also, `previousAppend, previousKey, obj` -> `parent, key, value`. More relevant names. Also, recursion typically does not care about "previous" values. What it needs is the data that applies to the "current" run.